### PR TITLE
Add logging of learning rates to tensorboard

### PR DESCRIPTION
**Pull request by matt-peters**
Add parameter specific learning rate logging to tensorboard.  It's optional, default is False.

### CODE REVIEW

**Comment by matt-gardner** [comment]
Does this replace what was done in #1641?

**Comment by matt-peters** [comment]
@DeNeutoy - agree, it would be nice to log the actual step sizes.

@matt-gardner - this will replace #1641 and an in a more general way (takes into consideration `requires_grad`) and in a more user friendly way (logs using the name of the parameter instead of an opaque parameter group index)
